File: Shared/java/code/threads/ThreadLocalTest.java
class ThreadLocalTest{

	static class PrintItem{

		private static String text;

		public static void write(String value){
			text = value;
		}

		public static String read(){
			return text;
		}
	}

	static class Printer{

		public static void print(int copies){
			for(int i = 1; i <= copies; ++i){
				System.out.printf("%d:%s%n", i, PrintItem.read());
				Worker.doWork(i);
			}
		}
	}

	public static void main(String[] args) throws Exception{
		PrintItem.write("monday");
		Printer.print(5);
		PrintItem.write("sunday");
		Printer.print(5);
	}
}

File: Shared/dotnet/code/threads/threadlocaltest.cs
using System;

static class PrintItem
{
	private static string text;

	public static void Write(string value)
	{
		text = value;
	}

	public static string Read()
	{
		return text;
	}
}

static class Printer
{
	public static void Print(int copies)
	{
		for(int i = 1; i <= copies; ++i)
		{
			Console.WriteLine("{0}:{1}", i, PrintItem.Read());
			Worker.DoWork(i);
		}
	}
}

static class Program
{
	public static void Main()
	{
		PrintItem.Write("Monday");
		Printer.Print(5);

		PrintItem.Write("Tuesday");
		Printer.Print(5);
	}
}